### PR TITLE
Add constructions for Monads and Comonads

### PR DESCRIPTION
**Pull request by 4e554c4c**
This commit adds a few constructions for monads and comonads. In particular, it adds the identity monad and comonad for a category. It also adds the identity and composite morphism between monads on the same category.

### CODE REVIEW

**Comment by JacquesCarette** [comment]
FYI: might be a few days before I can properly review this. Very welcome additions, I'm just facing a number of deadlines at the moment.

File: src/Categories/Comonad/Morphism.agda [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    ≈⟨ C.sym-assoc ⟩
+    (R.ε.η U ∘ σ .α.η U) ∘ τ .α.η  U
+    ≈⟨ σ .counit-comp ⟩∘⟨refl ⟩
+    T.ε.η U ∘ τ .α.η  U
+    ≈⟨ τ .counit-comp ⟩
+    S.ε.η U
+    ∎
+  Comonad⇒-id-∘ σ τ .comult-comp {U} = begin
+    R.δ.η U ∘ σ .α.η U ∘ τ .α.η U
+    ≈⟨ C.sym-assoc ⟩
```
Comment (JacquesCarette) [inline src/Categories/Comonad/Morphism.agda:122]: Various uses of push / pull / assoc2 etc can likely shorten this proof too.

File: src/Categories/Monad.agda [hunk 2]
```diff
@@ -21,10 +21,22 @@ record Monad {o ℓ e} (C : Category o ℓ e) : Set (o ⊔ ℓ ⊔ e) where
...
   field
     assoc     : ∀ {X : Obj} → μ.η X ∘ F₁ (μ.η X) ≈ μ.η X ∘ μ.η (F₀ X)
     sym-assoc : ∀ {X : Obj} → μ.η X ∘ μ.η (F₀ X) ≈ μ.η X ∘ F₁ (μ.η X)
     identityˡ : ∀ {X : Obj} → μ.η X ∘ F₁ (η.η X) ≈ id
     identityʳ : ∀ {X : Obj} → μ.η X ∘ η.η (F₀ X) ≈ id
+
+module _ {o ℓ e} (C : Category o ℓ e) where
+  open Monad
+  open Category C
+  idM : Monad C
```
Comment (JacquesCarette) [inline src/Categories/Monad.agda:35]: Just `id`

File: src/Categories/Monad/Morphism.agda [hunk 3]
```diff
@@ -2,71 +2,144 @@
...
-    comm : ∀ {U} → Δ.α.η U ∘ N.F.₁ (m.η U) ≈ m.η (M.F.₀ U) ∘ Γ.α.η U
\ No newline at end of file
+    module T = Monad T
+  open Monad⇒-id
+  open Category C
+  open HomReasoning
+  Monad⇒-id-id : (Monad⇒-id T T)
+  Monad⇒-id-id .α = idN
+  Monad⇒-id-id .unit-comp {_} = identityˡ
+  Monad⇒-id-id .mult-comp {U} = begin
```
Comment (JacquesCarette) [inline src/Categories/Monad/Morphism.agda:93]: Here and below: all the same comments as for Comonad apply.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Only some cosmetic changes, nice.

File: src/Categories/Comonad/Morphism.agda [hunk 4]
```diff
@@ -0,0 +1,142 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Level
+open import Categories.Category using (Category)
+open import Categories.Comonad
```
[the remark this answers is not in the document]
Comment (4e554c4c) [reply src/Categories/Comonad/Morphism.agda:5]: Here I believe I am using everything from a module (`Comonad` and `id`)

**Comment by 4e554c4c** [comment]
oops sorry for requesting review @Taneb  that was on accident

File: src/Categories/Comonad/Morphism.agda [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    module T = Comonad T
+  open Comonad⇒-id
+  open Category C
+  open HomReasoning
+  open MR C
+  Comonad⇒-id-id : (Comonad⇒-id T T)
+  Comonad⇒-id-id .α = idN
+  Comonad⇒-id-id .counit-comp {_} = identityʳ
+  Comonad⇒-id-id .comult-comp {U} = begin
+    T.δ.η U ∘ id
```
Comment (JacquesCarette) [inline src/Categories/Comonad/Morphism.agda:89]: If you don't want to do
```
lhs     reason
```
layout, please do at least 2-space indenting of the reasons.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Just one minor cosmetic thing left to fix (in a number of places). The rest is great - the proof-golfing really improved things.

**Review by JacquesCarette — DISMISSED** [review]
Perfect!

**Comment by JacquesCarette** [comment]
CI fails with
```
 Checking Categories.Monad.Strong (/home/runner/work/agda-categories/agda-categories/src/Categories/Monad/Strong.agda).
/home/runner/work/agda-categories/agda-categories/src/Categories/Monad/Strong.agda:47,43-45
Ambiguous name id. It could refer to any one of
  Categories.Monad.id bound at
    /home/runner/work/agda-categories/agda-categories/src/Categories/Monad.agda:35,3-5
  Categories.Monad.Strong.id bound at
    /home/runner/work/agda-categories/agda-categories/src/Categories/Category/Core.agda:23,5-7
id is in scope as
  * a defined name Categories.Monad._.id brought into scope by
    - the opening of Categories.Monad at /home/runner/work/agda-categories/agda-categories/src/Categories/Monad/Strong.agda:19,13-29
    - its definition at /home/runner/work/agda-categories/agda-categories/src/Categories/Monad.agda:35,3-5
  * a record field Categories.Monad.Strong._.id brought into scope by
    - the application of Category at /home/runner/work/agda-categories/agda-categories/src/Categories/Monad/Strong.agda:28,8-16
    - its definition at /home/runner/work/agda-categories/agda-categories/src/Categories/Category/Core.agda:23,5-7
when scope checking id
```

**Comment by 4e554c4c** [comment]
I ran the CI locally, it seems like everything should pass now

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**